app/tests/data_factory.py

```diff
@@ -0,0 +1,978 @@
+    model_class = Asset
+    sample_name = 'asset'
+    samples_dict = asset_samples
+    key_mapping = {
+        'name': 'id',
+        'eo_gsd': 'eo:gsd',
+        'geoadmin_variant': 'geoadmin:variant',
+        'geoadmin_lang': 'geoadmin:lang',
+        'proj_epsg': 'proj:epsg',
+        'media_type': 'type',
+        'checksum_multihash': 'checksum:multihash'
+    }
+    optional_fields = [
+        'title', 'description', 'eo_gsd', 'geoadmin_variant', 'geoadmin_lang', 'proj_epsg', 'file'
+    ]
+
+    def __init__(self, sample='asset-1', item=None, name=None, **kwargs):
+        '''Create a item sample data
+
+        Args:
+            sample: string
+                Name of the sample based to use, see tests.sample_data.asset_samples
+            required_only: bool
+                return only attributes that are required (minimum sample data)
+            item: Item
+                Item DB object relations
+            name: string
+                Overwrite the sample name
+            **kwargs:
+                any parameter will overwrite existing attributes
+        '''
+        self.attr_name = name
+        super().__init__(sample=sample, item=item, name=name, **kwargs)
+
+        file = getattr(self, 'attr_file', None)
+        if file:
+            self.attr_checksum_multihash = get_sha256_multihash(file)
+            self.attr_file = SimpleUploadedFile(
+                f'{item.collection.name}/{item.name}/{self.attr_name}', file
+            )
+
+    @property
+    def json(self):
+        '''Returns a json serializable representation of the sample data
+
+        This json payload can then be used in the write API payload or to check
+        the read API payload.
+
+        Returns
+            A dictionary with the sample data
+        '''
+        # pylint: disable=no-member
+        data = super().json
+        item = data.pop('item')
+        data['item'] = item.name
+        file = data.pop('file', None)
+        if file:
+            data['href'] = \
+                f'http://{settings.AWS_S3_CUSTOM_DOMAIN}/{item.collection.name}/{item.name}/{file.name}'
+        return data
+
+
+class FactoryBase:
+    '''Factory base class
+    '''
+    factory_name = 'base'
+    sample_class = SampleData
+
+    def __init__(self):
+        self.last = None
+
+    @classmethod
+    def get_last_name(cls, last):
+        '''Return a factory name incremented by one (e.g. 'collection-1')
+        '''
+        if last is None:
+            last = f'{cls.factory_name}-0'
+        last = '{}-{}'.format(
+            cls.factory_name, int(re.match(fr"{cls.factory_name}-(\d+)", last).group(1)) + 1
+        )
+        return last
+
+    def create_sample(self, name=None, db_create=False, **kwargs):
+        '''Create a data sample
+
+        Args:
+            name: string
+                Data name, if not given it creates a '{factory_name}-n' with n being incremented
+                after each function call
+            sample: string
```

Comment:
`sample` is not an argument of `create_sample` any longer. But `db_create` is one and is not described in the docstring here.

Reply:
Same as for `required_only` it is hidden in `**kwargs`